frontend/packages/dev-console/src/components/catalog/service/CatalogItemsLoader.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ResolvedExtension } from '@console/dynamic-plugin-sdk/src/api/useResolvedExtensions';
+import { CatalogExtensionHookOptions, CatalogItem, CatalogItemProvider } from '@console/plugin-sdk';
+import CatalogExtensionHookResolver from './CatalogExtensionHookResolver';
+
+type CatalogItemsLoaderProps = {
+  catalogType: string;
+  providerExtensions: ResolvedExtension<CatalogItemProvider>[];
+  options: CatalogExtensionHookOptions;
+  onItemsLoaded: (items: CatalogItem[], type: string) => void;
+  onLoadError: (error: any) => void;
+};
+
+const CatalogItemsLoader: React.FC<CatalogItemsLoaderProps> = ({
+  catalogType,
+  providerExtensions,
+  options,
+  onItemsLoaded,
+  onLoadError,
+}) => {
+  const [resolvedProviderValues, setResolvedProviderValues] = React.useState<{
+    [key: string]: CatalogItem[];
+  }>({});
+  const [loadError, setLoadError] = React.useState();
+
+  if (loadError) onLoadError(loadError);
```

Comment:
State seems unnecessary. 
Pass `onLoadError` to each resolver:
```
onValueError={onLoadError}
```

Reply:
Done